feat : change Webclient to RestTemplate

Hue-Jo opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
Webclient로 DB에 저장까지 완료했으나 어플리케이션을 종료하면 대기열 관련 문제 에러가 끊임없이 콘솔에 적히며 데이터 관련 작업이 끝날 떄까지 중단이 안 됨. RestTemplate로 변경했더니 문제없이 어플리케이션 종료 가능

**TO-BE**
데이터베이스에 데이터를 이미 저장한 상태인데 다시 Webclient로 바꿀 필요가 있을까요?

Hue-Jo committed:
Change Webclient to RestTemplate

Webclient로 DB에 저장까지 완료했으나 어플리케이션을 종료하면 대기열 관련 문제 에러가 끊임없이 콘솔에 적히며 데이터 관련 작업이 끝날 떄까지 중단이 안 됨.
RestTemplate로 변경했더니 문제없이 어플리케이션 종료 가능

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -36,90 +39,59 @@ public class MealServiceImpl implements MealService {
 
-    // DB에서 학교 정보 가져오기
-    List<School> schools = schoolRepository.findAll();
+    List<School> schools = schoolRepository.findAll(); // 저장된 학교 정보 가져오기
```
paging처리를 하지 않아도 될까요?

Hue-Jo replied:
추가하여 수정하겠습니다.

Hue-Jo replied:
@mingming-mentor 멘토님!  그런데 왜 페이징 처리를 데이터를 순회할 때 처리를 해야하는 건가요?

mingming-mentor replied:
@Hue-Jo 님 데이터를 전체 다 긁어오면 메모리 부족으로 OOM이 발생할 수 있습니다. 또한 DB에도 부하가 가구요. 따라서 페이징처리를 하여 데이터를 일정크기만큼 가지고 오면서 처리하셔야합니다

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -36,90 +39,59 @@ public class MealServiceImpl implements MealService {
-          if (responseBody == null) {
-            log.error("API로부터 데이터를 받아오는 데에 실패했습니다." +
-                "학교 코드: {}, 날짜: {}", schoolCode, formattedDate);
+          ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
```
String으로 가지고 오지 않고 response 형태에 맞는 class를 만드시어서 해당 class로 바로 convert 하면 좋을 것 같습니다. 
그러면 아래의 JsonNode등을 사요하지 않으셔도 되어 코드를 더 깔끔하게 유지할 수 있습니다

Hue-Jo replied:
@mingming-mentor 수정하여 커밋하였습니다. 뭔가... 말씀하신 그대로 된 것 같진 않긴한데...... 아무튼 데이터가 DB에 들어가긴합니다..!

mingming-mentor replied:
String.class 대신 MealResponse로 바로 받으셔도 될것 같네요. 그럼 불필요하게 objectMapper로 다시 맵핑하는 작업을 하지 않아도 될것 같습니다

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -36,90 +39,59 @@ public class MealServiceImpl implements MealService {
             API_URL, apiKey, 1, 100, EDU_OFFICE_CODE, schoolCode, formattedDate,
             formattedDate);
 
         try {
```
url은 UrlBuilder를 사용해주시면 좋을 것 같습니다

Hue-Jo replied:
확인했습니다. 확실히 가독성이 높아지네요!

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -36,90 +39,59 @@ public class MealServiceImpl implements MealService {
+            LocalDate mealDate = LocalDate.parse(node.path("MLSV_YMD").asText(),DATE_FORMAT);
+
+            // 기존 데이터와 중복을 피하기 위해 중복 검사
+            if (!mealRepository.existsBySchoolCodeAndMealDate(school, mealDate)) {
```
기존 데이터가 있더라도 업데이트가 될수도 있지 않을까요? 현재대로라면 업데이트는 불가능한대 의도하신 바가 맞나요?

Hue-Jo replied:
@mingming-mentor 
Optional로 기존 데이터가 있다면 업데이트할 필요가 있는지 확인하고 기존 데이터가 없다면 새로 추가하는 방식으로 수정하겠습니다.

mingming-mentor reviewed (approved):
webClient 사용시 발생하셨던 문제에 대해서 같이 봐주시길 원하시는거라면 어떤 문제가 있었는지를 같이 기재해주셔야 제가 알수 있을 것 같아요. 
현재 말씀주신 상태에서는 webclient는 원하시는대로 동작을 안하고 restTemplate은 원하시는대로 동작을 하는 것으로 이해했습니다. 원인 파악이 가능하다면 원인 파악을 하시면 좋을 것 같고, 어렵다면 사용은 restTemplate으로만 가능하니 restTemplate을 사용하시는 것이 맞을 것 같네요. 
또한 두 기술에 대해서 한번 공부해보시면 현재 구조상 어떤 기술이 좀더 적합할지 판단이 가실거라 생각이 듭니다. 단순히 같은 기술을 구현한 라이브러리가 아니라, 다른 특성을 가진 라이브러리이고 해당 특성이 큰 요소이기 때문에 판단의 기준으로 삼을 수가 있습니다.

mingming-mentor commented:
이쪽 PR에 수정을 하시는 것이 제일 좋으며, 이쪽 PR은 머지하신 뒤에 이후 새로운 PR에서 피드백 반영하셔도 됩니다. 참고삼으시라고 말씀드려요

Hue-Jo committed:
Change based on PR

Response.class 추가,
Optional로 기존 데이터가 있다면 업데이트할 필요가 있는지 확인하고 기존 데이터가 없다면 새로 추가하는 방식으로 수정
